### ISSUE TITLE
Analysis memory consuption explodes on a few binaries

### ISSUE THREAD

**Issue by janfrancu** [Vector35/binaryninja-api#991]
I have encountered unexpected memory consumption when analyzing following PE binaries: `8900c5963a090b8607ee072901b23af30db0d88d3807f1cb28576e61627430cc`,  `f1699a30447280df50f730f469223cf1f8aca1e23c0a5904c61b9f3a079a2b05`, `81ed49f61026d2335ccc7cfe8a7c269b4faaf04baa747242a8c3454497212c0c`.
`9a396ac1592be38f0d70948e627275ff32b52dfb026085032437dd1f8b399140`.

Using the dev version, the analysis in UI completes and reports that there are function too large to analyze and offers force analysis, running it makes the memory consumption explode. From my brief look, it seems like the analysis gets sidetracked into data sections (at least for the first three hashes), where it finds functions in null bytes.

Running analysis from the API follows almost the same pattern. The analysis completes, but when accessing blocks of the afflicted function, it freezes and memory consumption explodes. The minimal script to reproduce is following

```python
bv = bn.binaryview.BinaryViewType['PE'].open(filename)
bv.update_analysis_and_wait()
for f in bv.functions:
	for b in f.basic_blocks:
		dtext = b.disassembly_text
```
If not possible to fix directly, I would appreciate the ability to query the API, if the function is too large (even though it might not be the case, as it probably should not have been discovered in the first place) to analyze so that I can avoid crashing the script.

**Comment by D0ntPanic** [comment]
You can check if the analysis thinks the function is too large by using the `too_large` property on the function object.

**Comment by bpotchik** [comment]
@janfrancu, the most recent build, 1.1.1446-dev, contains improved analysis for PE binaries. Would it be possible to check if these binaries are still causing an issue? Thanks!

**Comment by janfrancu** [comment]
Running them with 1.1.1448-dev still yields the same behavior. I forgot to mention in my post that those files can be fetched from Virus Total. I am attaching them now, so that you can debug it.
[exploding.zip](https://github.com/Vector35/binaryninja-api/files/2653001/exploding.zip)
I don't know if it has been reported before, but note that there is now also an issue with one of the files (`f1699a30447280df50f730f469223cf1f8aca1e23c0a5904c61b9f3a079a2b05`), where Binja creates functions in the PE header.

**Comment by bpotchik** [comment]
Ah, thanks for the binaries. I found the issue and will have a fix out shortly.

**Comment by bpotchik** [comment]
Okay, I fixed several issues in 1.1.1450-dev. It appears that all the 'explosions' are fixed. Although it appears that 9a396ac1592be38f0d70948e627275ff32b52dfb026085032437dd1f8b399140 is obfuscated or packed as there are many bad functions being created during Phase 3 (Control Flow Graph Recovery) analysis. You can always disable linear sweep for that specific binary for now.

**Comment by bpotchik** [comment]
The remaining issues have been fixed in 1.1.1578-dev.
